--- a/frameworks/base/telephony/java/android/telephony/SignalStrength.java
+++ b/frameworks/base/telephony/java/android/telephony/SignalStrength.java
@@ -854,6 +854,9 @@
          * RSRQ = quality of signal dB = Number of Resource blocks*RSRP/RSSI
          * SNR = gain = signal/noise ratio = -10log P1/P2 dB
          */
+	String GetRilStrength = android.os.SystemProperties.get("ril.signal.param");
+	String RilStrength[] = GetRilStrength.split(",");
+
         int rssiIconLevel = SIGNAL_STRENGTH_NONE_OR_UNKNOWN, rsrpIconLevel = -1, snrIconLevel = -1;
 
         if (mLteRsrp > MAX_LTE_RSRP || mLteRsrp < MIN_LTE_RSRP) {
@@ -923,7 +926,7 @@
         // TODO the range here is probably supposed to be (0..31, 99). It's unclear if anyone relies
         // on the current incorrect range check, so this will be fixed in a future release with more
         // soak time
-        if (mLteSignalStrength > 63) rssiIconLevel = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
+        /*if (mLteSignalStrength > 63) rssiIconLevel = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
         else if (mLteSignalStrength >= 12) rssiIconLevel = SIGNAL_STRENGTH_GREAT;
         else if (mLteSignalStrength >= 8) rssiIconLevel = SIGNAL_STRENGTH_GOOD;
         else if (mLteSignalStrength >= 5) rssiIconLevel = SIGNAL_STRENGTH_MODERATE;
@@ -931,7 +934,15 @@
 
         if (DBG) log("getLteLevel - rssi:" + mLteSignalStrength + " rssiIconLevel:"
                 + rssiIconLevel);
-        return rssiIconLevel;
+        return rssiIconLevel;*/
+
+	if (Integer.parseInt(RilStrength[1]) > 63) rssiIconLevel = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
+	else if (Integer.parseInt(RilStrength[1]) >= 10) rssiIconLevel = SIGNAL_STRENGTH_GREAT;
+	else if (Integer.parseInt(RilStrength[1]) >= 6) rssiIconLevel = SIGNAL_STRENGTH_GOOD;
+        else if (Integer.parseInt(RilStrength[1]) >= 0) rssiIconLevel = SIGNAL_STRENGTH_MODERATE;
+        else if (Integer.parseInt(RilStrength[1]) >= -7) rssiIconLevel = SIGNAL_STRENGTH_POOR;
+
+	return rssiIconLevel;
 
     }
 
